gap/SubfieldMatrixGroups.gi

```diff
@@ -278,3 +278,45 @@ function(epsilon, d, q)
  
     return result;
 end);
+
+
+# Construction as in Proposition 8.2 of [HR05]
+BindGlobal("SubfieldSp",
+function (d, p, e, f)
+    local F, q0, b, gens, l, zeta, omega, zetaPower, C, gen;
+
+    if IsOddInt(d) then
+    	ErrorNoReturn("<d> must even but ", d, " was given.");
+    fi;
+
+    F := GF(p ^ e);
+    q0 := p ^ f;
+    b := QuoInt(e, f);
+    gens := List(GeneratorsOfGroup(Sp(d, q0)));
+
+    if e mod f <> 0 or not IsPrime(b) then
+        ErrorNoReturn("<f> must be a divisor of <e> and their quotient must be a prime but <e> = ", 
+                      e, " and <f> = ", f);
+    fi;
```

Comment:
In general I'd try move input validation to the very start, before any work is done. Also, this checks for two conditions, and so I'd suggest to have two errors; that makes it slightly easier for a human to understand what's going on. I.e., together with what I wrote above:
```suggestion
    if e mod f <> 0 then
        ErrorNoReturn("<f> must be a divisor of <e>");
    fi;
    if not IsPrime(b) then
        ErrorNoReturn("The quotient of <f> by <e> must be a prime");
    fi;
```